crates/typed-store/src/raw_key_db_map.rs

```diff
@@ -0,0 +1,498 @@
+// Copyright (c) Walrus Foundation
+// SPDX-License-Identifier: Apache-2.0
+
+//! Custom DBMap wrapper with configurable key serialization/deserialization.
+//!
+//! This module provides a wrapper around typed_store's DBMap that allows
+//! for custom key serialization/deserialization strategies, enabling
+//! optimized key encoding for specific use cases, like to preserve alphanumeric order.
+
+use std::marker::PhantomData;
+
+use rocksdb::OptimisticTransactionDB;
+use serde::{Deserialize, Serialize};
+
+use crate::{TypedStoreError, rocks::DBMap};
+
+/// Trait for key serialization and deserialization.
+pub trait KeyCodec: Sized {
+    /// Serialize the key to bytes.
+    fn serialize(&self) -> Result<Vec<u8>, TypedStoreError>;
+
+    /// Deserialize a key from bytes.
+    fn deserialize(bytes: &[u8]) -> Result<Self, TypedStoreError>;
+}
+
+/// A wrapper around DBMap that allows raw key serialization/deserialization.
+///
+/// This struct provides a minimal interface for using DBMap with raw
+/// key serialization/deserialization while keeping the default BCS serialization for values.
+pub struct RawKeyDBMap<K, V>
+where
+    K: KeyCodec,
+    V: Serialize + for<'de> Deserialize<'de> + Clone,
+{
+    inner: DBMap<Vec<u8>, V>,
+    _phantom: PhantomData<K>,
+}
```

Comment:
question: is the future plan to implement the full [Map trait](https://github.com/MystenLabs/walrus/blob/ea3735eb08f63abc44f3e4a6e8d8bfc34f08c3fe/crates/typed-store/src/traits.rs#L11) in RawKeyDBMap as well? Right now, although inner is DBMap, the implementation only uses inner to access the underlying RocksDB column family, and directly operate on that. If the usage remain like this, I wonder if the `inner `can just be the column family handle instead of a DBMap. It doesn't really need to use DBMap.

Also, I feel it makes things a bit more unintuitive to "untype" the key on top of DBMap (whose original design is just type the raw byte key and value, and help with serialization and deserialization).